File: admin/Skins/Links.ascx.vb
'
' DotNetNuke® - http://www.dotnetnuke.com
' Copyright (c) 2002-2007
' by DotNetNuke Corporation
'
' Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
' documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
' the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
' to permit persons to whom the Software is furnished to do so, subject to the following conditions:
'
' The above copyright notice and this permission notice shall be included in all copies or substantial portions 
' of the Software.
'
' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
' TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
' THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
' CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
' DEALINGS IN THE SOFTWARE.
'
Imports DotNetNuke.Entities.Tabs

Namespace DotNetNuke.UI.Skins.Controls

    ''' -----------------------------------------------------------------------------
    ''' <summary></summary>
    ''' <remarks></remarks>
    ''' <history>
    ''' 	[cniknet]	10/15/2004	Replaced public members with properties and removed
    '''                             brackets from property names
    ''' </history>
    ''' -----------------------------------------------------------------------------
    Partial Class Links
        Inherits UI.Skins.SkinObjectBase

        ' private members
        Private _separator As String
        Private _cssClass As String
        Private _level As String
        Private _alignment As String

#Region "Public Members"
        Public Property Separator() As String
            Get
                Return _separator
            End Get
            Set(ByVal Value As String)
                _separator = Value
            End Set
        End Property

        Public Property CssClass() As String
            Get
                Return _cssClass
            End Get
            Set(ByVal Value As String)
                _cssClass = Value
            End Set
        End Property

        Public Property Level() As String
            Get
                Return _level
            End Get
            Set(ByVal Value As String)
                _level = Value
            End Set
        End Property

        Public Property Alignment() As String
            Get
                Return _alignment
            End Get
            Set(ByVal Value As String)
                _alignment = Value
            End Set
        End Property

#End Region

        Private Sub Page_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

            ' public attributes
            Dim strCssClass As String
            If CssClass <> "" Then
                strCssClass = CssClass
            Else
                strCssClass = "SkinObject"
            End If

            Dim strSeparator As String
            If Separator <> "" Then
                If Separator.IndexOf("src=") <> -1 Then
                    strSeparator = Replace(Separator, "src=", "src=" & PortalSettings.ActiveTab.SkinPath)
                Else
                    strSeparator = "<span class=""" & strCssClass & """>" & Replace(Separator, " ", "&nbsp;") & "</span>"
                End If
            Else
                strSeparator = "  "
            End If

            ' build links
            Dim strLinks As String = ""

            strLinks = BuildLinks(Level, Alignment, strSeparator, strCssClass)

            If strLinks = "" Then
                strLinks = BuildLinks("", Alignment, strSeparator, strCssClass)
            End If

            lblLinks.Text = strLinks

        End Sub

        Private Function BuildLinks(ByVal Level As String, ByVal Alignment As String, ByVal strSeparator As String, ByVal strCssClass As String) As String

            Dim strLinks As String = ""
            Dim strLoop As String
            Dim intIndex As Integer

            For intIndex = 0 To PortalSettings.DesktopTabs.Count - 1

                Dim objTab As TabInfo = CType(PortalSettings.DesktopTabs(intIndex), TabInfo)

                If objTab.IsVisible = True And objTab.IsDeleted = False And objTab.DisableLink = False Then
                    If (objTab.StartDate < Now And objTab.EndDate > Now) Or AdminMode = True Then
                        If PortalSecurity.IsInRoles(objTab.AuthorizedRoles) = True Then
                            strLoop = ""
                            If Alignment = "Vertical" Then
                                If strLinks <> "" Then
                                    strLoop = "<br />" & strSeparator
                                Else
                                    strLoop = strSeparator
                                End If
                            Else
                                If strLinks <> "" Then
                                    strLoop = strSeparator
                                End If
                            End If

                            Select Case Level
                                Case "Same", ""
                                    If objTab.ParentId = PortalSettings.ActiveTab.ParentId Then
                                        strLinks += strLoop & AddLink(objTab.TabName, objTab.FullUrl, strCssClass)
                                    End If
                                Case "Child"
                                    If objTab.ParentId = PortalSettings.ActiveTab.TabID Then
                                        strLinks += strLoop & AddLink(objTab.TabName, objTab.FullUrl, strCssClass)
                                    End If
                                Case "Parent"
                                    If objTab.TabID = PortalSettings.ActiveTab.ParentId Then
                                        strLinks += strLoop & AddLink(objTab.TabName, objTab.FullUrl, strCssClass)
                                    End If
                                Case "Root"
                                    If objTab.Level = 0 Then
                                        strLinks += strLoop & AddLink(objTab.TabName, objTab.FullUrl, strCssClass)
                                    End If
                            End Select
                        End If
                    End If
                End If

            Next intIndex

            Return strLinks

        End Function

        Private Function AddLink(ByVal strTabName As String, ByVal strURL As String, ByVal strCssClass As String) As String

            Return "<a class=""" & strCssClass & """ href=""" & strURL & """>" & strTabName & "</a>"

        End Function

    End Class

End Namespace

File: DesktopModules/Store/SkinObjects/Links.ascx.cs
/*
'  DotNetNuke -  http://www.dotnetnuke.com
'  Copyright (c) 2002-2007
'  by Shaun Walker ( sales@perpetualmotion.ca ) of Perpetual Motion Interactive Systems Inc. ( http://www.perpetualmotion.ca )
' 
'  Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
'  documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
'  the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
'  to permit persons to whom the Software is furnished to do so, subject to the following conditions:
' 
'  The above copyright notice and this permission notice shall be included in all copies or substantial portions 
'  of the Software.
' 
'  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
'  TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
'  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
'  CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
'  DEALINGS IN THE SOFTWARE.
*/

using System;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.HtmlControls;
using DotNetNuke;
using DotNetNuke.Common;
using DotNetNuke.Services.Localization;
using DotNetNuke.Modules.Store.Admin;

namespace DotNetNuke.Modules.Store.SkinObjects
{
    public partial class Links : DotNetNuke.UI.Skins.SkinObjectBase
    {

        #region Web Form Designer generated code
        override protected void OnInit(EventArgs e)
        {
            InitializeComponent();
            base.OnInit(e);
        }

        private void InitializeComponent()
        {
        }
        #endregion

        #region Properties
        private string _imageCssClass = "Normal";
        private string _imageName = "cart.png";
        private string _linkAction = "Cart";
        private string _textCssClass = "Normal";
        private bool _textVisible = true;

        public string ImageCssClass
        {
            get
            {
                return _imageCssClass;
            }
            set
            {
                _imageCssClass = value;
            }
        }

        public string ImageName
        {
            get
            {
                return _imageName;
            }
            set
            {
                _imageName = value;
            }
        }

        public string LinkAction
        {
            get
            {
                return _linkAction;
            }
            set
            {
                _linkAction = value;
            }
        }

        public string TextCssClass
        {
            get
            {
                return _textCssClass;
            }
            set
            {
                _textCssClass = value;
            }
        }

        public bool TextVisible
        {
            get
            {
                return _textVisible;
            }
            set
            {
                _textVisible = value;
            }
        }
        #endregion
        
        #region event Handlers
        protected override void OnPreRender(EventArgs e)
        {
            base.OnPreRender(e);

            if (!IsPostBack)
            {
                string _resource = this.TemplateSourceDirectory + "/App_LocalResources/Links.ascx.resx";
                string _text = "";

                try
                {
                    StoreController storeController = new StoreController();
                    StoreInfo storeInfo = storeController.GetStoreInfo(PortalSettings.PortalId);

                    int _tabID = 0;

                    switch (_linkAction.ToLower())
                    {
                        case "cart":
                            _text = Localization.GetString("CartTitle.Text", _resource);
                            _tabID = storeInfo.ShoppingCartPageID;
                            break;
                        case "store":
                            _text = Localization.GetString("StoreTitle.Text", _resource);
                            _tabID = storeInfo.StorePageID;
                            break;
                        default:
                            _text = Localization.GetString("UnknowAction.Text", _resource);
                            _tabID = PortalSettings.HomeTabId;
                            _textVisible = true;
                            break;
                    }

                    if (_textVisible)
                    {
                        lnkAction.CssClass = _textCssClass;
                        lnkAction.Text = _text;
                        lnkAction.PostBackUrl = Globals.NavigateURL(_tabID);
                    }
                    lnkAction.Visible = _textVisible;

                    if (_imageName != string.Empty)
                    {
                        if (storeInfo.PortalTemplates)
                        {
                            btnImage.ImageUrl = PortalSettings.HomeDirectory + "Store/Templates/Images/" + _imageName;
                        }
                        else
                        {
                            btnImage.ImageUrl = this.TemplateSourceDirectory + "/../Templates/Images/" + _imageName;
                        }
                        btnImage.CssClass = _imageCssClass;
                        btnImage.ToolTip = _text;
                        btnImage.PostBackUrl = Globals.NavigateURL(_tabID);
                    }
                    else
                    {
                        btnImage.Visible = false;
                    }
                }
                catch
                {
                    lnkAction.CssClass = _textCssClass;
                    _text = Localization.GetString("Error.Text", _resource);
                    lnkAction.Text = _text;
                    btnImage.Visible = false;
                }
            }
        }
        #endregion
    }
}
